lib/node_modules/@stdlib/string/base/last-code-point/lib/main.js

```diff
@@ -0,0 +1,99 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2024 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var RE_UTF16_SURROGATE_PAIR = require( '@stdlib/regexp/utf16-surrogate-pair' ).REGEXP;
+
+
+// VARIABLES //
+
+var RE_UTF16_LOW_SURROGATE = /[\uDC00-\uDFFF]/; // TODO: replace with stdlib pkg
+var RE_UTF16_HIGH_SURROGATE = /[\uD800-\uDBFF]/; // TODO: replace with stdlib pkg
+
+
+// MAIN //
+
+/**
+* Returns the last `n` Unicode code points of a string.
+*
+* @param {string} str - input string
+* @param {NonNegativeInteger} n - number of Unicode code points to return
+* @returns {string} output string
+*
+* @example
+* var out = lastCodePoint( 'Hello World', 1 );
+* // returns 'd'
+*
+* @example
+* var out = lastCodePoint( 'ASCII', 2 );
+* // returns 'II'
+*
+* @example
+* var out = lastCodePoint( 'JavaScript', 6 );
+* // returns 'Script'
+*
+*/
+function lastCodePoint( str, n ) {
+	var len;
+	var out;
+	var ch1;
+	var ch2;
+	var cnt;
+	var i;
+	len = str.length;
+	out = '';
+	cnt = 0;
+	if ( str === '' || n === 0 ) {
+		return '';
+	}
+	if ( n === 1 ) {
+		str = str.substring( len-2, len );
+		if ( RE_UTF16_SURROGATE_PAIR.test( str ) ) {
+			return str;
+		}
+		return str[1];
+	}
+	for ( i = len-1; i >= 0; i-- ) {
+		ch1 = str[ i ];
+		out = ch1 + out;
+		cnt += 1;
+
+		if ( RE_UTF16_LOW_SURROGATE.test( ch1 ) ) {
+			if ( i === 0 ) {
+				break;
+			}
+			ch2 = str[ i+1 ];
+			if ( RE_UTF16_HIGH_SURROGATE.test( ch2 ) ) {
+				out += ch2;
+				i -= 1;
+			}
+		}
+		if ( cnt === n ) {
+			break;
+		}
+	}
+	return out;
+}
```

Comment:
I'll recommend adding comments as in https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/string/base/first-code-point/lib/main.js#L78

Reply:
ill do that